crates/ruff_linter/src/rules/flake8_async/rules/cancel_scope_no_checkpoint.rs

```diff
@@ -80,6 +83,17 @@ pub(crate) fn cancel_scope_no_checkpoint(
         return;
     }
 
+    // Treat yields as checkpoints, since checkpoints can happen
+    // in the caller yielded to.
+    // https://flake8-async.readthedocs.io/en/latest/rules.html#async100
+    // https://github.com/astral-sh/ruff/issues/12873
+    if with_stmt.body.iter().any(|stmt| {
```

Comment:
The logic here only traverses the direct children of a with statement but, for example, not the statements in an if body. Is this intentional? If not, consider using the StatementVisitor

Reply:
Not intentional, thank you! I tried using `any_over_body` instead. Unless I'm misunderstanding, that's slightly faster since it gets to stop traversing the tree as soon as a `yield` is found, right?

Reply:
I'm not familiar with that method. I'll take a look tomorrow

Reply:
Yup, correct usage of `any_over_body`!